tests/sentry/integrations/slack/test_utils.py

```diff
@@ -103,3 +103,193 @@ def test_rate_limiting(self):
+        self.add_msg_response("my-channel", "m-c")
+        self.resp.add(
+            method=responses.POST,
+            url="https://slack.com/api/chat.deleteScheduledMessage",
+            status=200,
+            content_type="application/json",
+            body=json.dumps({"ok": True}),
+        )
+        self.run_valid_test("#My-Channel", CHANNEL_PREFIX, "m-c", False)
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def test_valid_private_channel_selected_new(self):
+        self.add_msg_response("my-private-channel", "m-p-c")
+        self.resp.add(
+            method=responses.POST,
+            url="https://slack.com/api/chat.deleteScheduledMessage",
+            status=200,
+            content_type="application/json",
+            body=json.dumps({"ok": True}),
+        )
+        self.run_valid_test("#my-private-channel", CHANNEL_PREFIX, "m-p-c", False)
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def test_valid_member_selected(self):
+        # add_msg_response making sure no match comes back so you fall into the user.list code path
+        # mock the user.list response so you find your user
+        self.add_bad_msg_response()
+        self.add_list_response(
+            "users",
+            [
+                {"name": "first-morty", "id": "m", "profile": {"display_name": "Morty"}},
+                {"name": "other-user", "id": "o-u", "profile": {"display_name": "Jimbob"}},
+                {"name": "better_morty", "id": "bm", "profile": {"display_name": "Morty"}},
+            ],
+            result_name="members",
+        )
+        self.run_valid_test("@first-morty", MEMBER_PREFIX, "m", False)
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def test_valid_member_selected_display_name(self):
+        self.add_bad_msg_response()
+        self.add_list_response(
+            "users",
+            [
+                {"name": "first-morty", "id": "m", "profile": {"display_name": "Morty"}},
+                {"name": "other-user", "id": "o-u", "profile": {"display_name": "Jimbob"}},
+                {"name": "better_morty", "id": "bm", "profile": {"display_name": "Morty"}},
+            ],
+            result_name="members",
+        )
+        self.run_valid_test("@Jimbob", MEMBER_PREFIX, "o-u", False)
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def test_invalid_member_selected_display_name(self):
+        self.add_bad_msg_response()
+        with pytest.raises(DuplicateDisplayNameError):
+            get_channel_id(self.organization, self.integration, "@Morty")
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def test_invalid_channel_selected(self):
+        self.add_bad_msg_response()
+        assert get_channel_id(self.organization, self.integration, "#fake-channel")[1] is None
+        assert get_channel_id(self.organization, self.integration, "@fake-user")[1] is None
+
+
+class GetChannelIdFasterErrorTest(TestCase):
```

Comment:
I know the existing tests are broken up into 2 classes but it's not necessary, if you combine them you don't need lines 254-281.